test/v5/createAmendUkPropertyCumulativeSummary/CreateAmendUkPropertyCumulativeSummaryServiceSpec.scala

```diff
@@ -0,0 +1,149 @@
+          )
+        )
+      )
+    )
+
+  val responseBodyJson: JsValue = Json.parse(
+    s"""
+       |{
+       |  "submissionId":"4557ecb5-fd32-48cc-81f5-e6acd1099f3c"
+       |}
+       """.stripMargin
+  )
+
+  val responseData: CreateAmendUkPropertyCumulativeSummaryResponse =
+    CreateAmendUkPropertyCumulativeSummaryResponse(submissionId = submissionId)
+
+  "CreateAmendUkPropertyCumulativeSummaryService" when {
+    "downstream call is successful" when {
+      "a submission id is returned from downstream" must {
+        "return a successful result" in new Test {
+          MockedCreateUkPropertyCumulativeSummaryConnector.createAmendUkPropertyCumulativeSummary(requestData) returns
+            Future.successful(Right(ResponseWrapper(correlationId, responseData)))
+
+          await(service.createAmendUkPropertyCumulativeSummary(requestData)) shouldBe Right(ResponseWrapper(correlationId, responseData))
+        }
+      }
+    }
+
+    "downstream call is unsuccessful" should {
+      "map errors according to spec" when {
+
+        def serviceError(downStreamErrorCode: String, error: MtdError): Unit =
+          s"a $downStreamErrorCode error is returned from the service" in new Test {
+            MockedCreateUkPropertyCumulativeSummaryConnector.createAmendUkPropertyCumulativeSummary(requestData) returns
+              Future.successful(Left(ResponseWrapper(correlationId, DownstreamErrors.single(DownstreamErrorCode(downStreamErrorCode)))))
+
+            await(service.createAmendUkPropertyCumulativeSummary(requestData)) shouldBe Left(ErrorWrapper(correlationId, error))
+          }
+
+        val errorMap = List(
+          "INVALID_TAXABLE_ENTITY_ID"          -> NinoFormatError,
+          "INVALID_INCOME_SOURCE_ID"           -> BusinessIdFormatError,
+          "INVALID_PAYLOAD"                    -> InternalError,
+          "INVALID_CORRELATION_ID"             -> InternalError,
+          "INVALID_TAX_YEAR"                   -> TaxYearFormatError,
+          "INCOME_SOURCE_NOT_FOUND"            -> NotFoundError,
+          "INCOME_SOURCE_DATA_NOT_FOUND"       -> NotFoundError,
+          "MISSING_EXPENSES"                   -> InternalError,
+          "INVALID_SUBMISSION_END_DATE"        -> RuleInvalidSubmissionEndDateError,
+          "SUBMISSION_END_DATE_VALUE"          -> RuleSubmissionEndDateError,
+          "INVALID_START_DATE"                 -> RuleStartDateNotAlignedWithReportingType,
+          "START_DATE_NOT_ALIGNED"             -> RuleStartDateNotAlignedToCommencementDate,
+          "END_DATE_NOT_ALIGNED"               -> RuleEndDateNotAlignedWithReportingType,
+          "MISSING_SUBMISSION_DATES"           -> RuleMissingSubmissionDates,
+          "START_END_DATE_NOT_ACCEPTED"        -> RuleStartAndEndDateNotAllowed,
+          "OUTSIDE_AMENDMENT_WINDOW"           -> RuleOutsideAmendmentWindow,
+          "TAX_YEAR_NOT_SUPPORTED"             -> RuleTaxYearNotSupportedError,
```

Comment:
Not sure why they do this but we also need to add SUBMITTED_TAX_YEAR_NOT_SUPPORTED -> `RuleTaxYearNotSupportedError` as well as TAX_YEAR_NOT_SUPPORTED.